File: app/src/main/java/com/drdisagree/iconify/services/BootReceiver.java
package com.drdisagree.iconify.services;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.drdisagree.iconify.BuildConfig;

public class BootReceiver extends BroadcastReceiver {

    private final String TAG = getClass().getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            if (BuildConfig.DEBUG) {
                Log.i(TAG, "Broadcast received: " + intent.getAction());
            }
            UpdateScheduler.scheduleUpdates(context);
        }
    }
}

File: app/src/main/java/com/drdisagree/iconify/services/receivers/BootReceiver.kt
package com.drdisagree.iconify.services.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.drdisagree.iconify.services.schedulers.UpdateScheduler

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            UpdateScheduler.scheduleUpdates(context)
        }
    }
}
